src/cic/rscript.clj

```diff
@@ -4,17 +4,18 @@
             [clj-time.format :as f]
             [clojure.java.shell :as shell]
             [clojure.set :as cs]
-            [clojure.string :as str]))
+            [clojure.string :as str]
+            [taoensso.timbre :as log]))
 
 (defn exec
   [script-path & args]
-  (println (format "Executing %s %s" script-path (str/join " " args)))
+  (log/info (format "Executing %s %s" script-path (str/join " " args)))
   (let [return-val (apply shell/sh "Rscript" "--vanilla" script-path args)]
     ;; rscript is quite chatty, so only pass on err text if exit was abnormal
-    (when (not= 0 (:exit return-val))
-      (apply println script-path args)
-      (println (:err return-val)))
-    (println (format "Executed %s" script-path))))
+    (if (not= 0 (:exit return-val))
+      (do (log/error script-path args)
+          (log/error return-val))
```

Comment:
Does this want to throw an ex-info exception or is it OK to just carry on?

Reply:
I reckon if any of the iterations fails to execute R properly then that's probably an irrecoverable error and we should probably throw an exception 👍